### PR TITLE
Have circuit breaker succeed on unknown mem usage

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we implement a workaround for
https://bugs.openjdk.java.net/browse/JDK-8207200 which is a race
condition in the JVM that results in `IllegalArgumentException` to be
thrown in rare cases when we determine memory usage via `MemoryMXBean`.
As we do not want to fail requests in those cases we always return zero
memory usage.

Relates #31767

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java [hunk 1]
```diff
@@ -251,7 +251,15 @@ private ParentMemoryUsage parentUsed(long newBytesReserved) {
 
     //package private to allow overriding it in tests
     long currentMemoryUsage() {
-        return MEMORY_MX_BEAN.getHeapMemoryUsage().getUsed();
+        try {
+            return MEMORY_MX_BEAN.getHeapMemoryUsage().getUsed();
+        } catch (IllegalArgumentException ex) {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java]: Neither of the methods here document that they can throw an `IllegalArgumentException`. So this catch looks quite odd and now I wonder if it's too broad. Catching it is fine, but can also check the exception message and `assert` that it is what we expect it to be? I don't want us to end up suppressing anything else here.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a suggestion.

File: server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java (hunk 1 above)
Comment (danielmitterdorfer) [reply server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java]: The `IllegalArgumentException` is thrown by the  [constructor of `MemoryUsage`](http://hg.openjdk.java.net/jdk/jdk11/file/1ddf9a99e4ad/src/java.management/share/classes/java/lang/management/MemoryUsage.java#l165). It does contain a few more precondition checks but I can add an assert. However, this we need to resort to pattern matching as the message contains absolute numbers, e.g:

```
committed = 542113792 should be < max = 536870912
```

Comment (danielmitterdorfer) [reply server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java]: Ah ok. Then I just misunderstood. Anyway, I can implement the assertion as suggested.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Review by dakrone — APPROVED** [review]
LGTM, thanks Daniel

**Pull request merged**